extensions/amp-3d-gltf/0.1/amp-3d-gltf.js

```diff
@@ -143,8 +143,9 @@ export class Amp3dGltf extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
-    observeWithSharedInOb(this.element, (inViewport) =>
-      this.viewportCallback_(inViewport)
+    this.unobserveIntersections_ = observeIntersections(
+      this.element,
+      (inViewport) => this.viewportCallback_(inViewport)
```

Comment:
Since callbacks are now called with the `IntersectionObserverEntry`, we should destructure in the param:
```suggestion
      ({isIntersecting}) => this.viewportCallback_(inViewport)
```

Reply:
Shouldn't we also need to check if `isIntersecting` has changed against an internal state, similar to BentoIframe?
https://github.com/ampproject/amphtml/blob/71f17a6aab18ce424e0d19185d2f6de39d6b4729/extensions/amp-iframe/1.0/component.js#L111-L113

Reply:
+1 re. destructuring, this'll break as-is.

> Shouldn't we also need to check if isIntersecting has changed against an internal state, similar to BentoIframe?

Yes, but I wouldn't require that in this PR since it is an already existing bug right?
It is a fairly unlikely situation, and the problem that arises is also not an issue (i.e. running the side effects of viewportCalback=true twice in a row or false twice in a row should both be fine)

Reply:
Done with destructuring changes.

> Shouldn't we also need to check if isIntersecting has changed against an internal state, similar to BentoIframe?

>> Yes, but I wouldn't require that in this PR since it is an already existing bug right?
It is a fairly unlikely situation, and the problem that arises is also not an issue (i.e. running the side effects of viewportCalback=true twice in a row or false twice in a row should both be fine)

Correct, since this is an existing bug not making any changes re: this.